webui/react/src/e2e/models/BaseComponent.ts

```diff
@@ -0,0 +1,83 @@
+import { type Locator } from '@playwright/test';
+import { BasePage } from './BasePage';
+
+/**
+ * Used in the constructor for BaseComponent
+ */
+export interface BaseComponentProps {
+  parent: BasePage | BaseComponent
+  selector?: string
+}
+
+/**
+ * Returns the representation of a Component.
+ * 
+ * @remarks
+ * This constructor is a base class for any component in src/components/.
+ * 
+ * @param {Object} obj
+ * @param {BasePage | BaseComponent} obj.parent - The parent used to locate this BaseComponent
+ * @param {string} [obj.selector] - Used instead of `defaultSelector`
+ */
+export class BaseComponent {
+  readonly defaultSelector: undefined | string;
+
+  readonly #selector: string;
+  protected readonly _parent: BasePage | BaseComponent;
+  protected _locator: Locator | undefined;
+
+  constructor({ parent, selector }: BaseComponentProps) {
+    if (typeof this !== typeof BaseComponent && typeof this.defaultSelector === 'undefined') {
+      throw new Error(`defaultSelector is undefined in class ${typeof this}`);
```

Comment:
If an undefined default selector is an error, can we make it a `string` instead of a `undefined | string`. Let the compiler do the null checking for us.